packages/core/src/browser/messaging/ws-connection-provider.ts

```diff
@@ -64,6 +64,7 @@ export class WebSocketConnectionProvider extends AbstractConnectionProvider<WebS
             this.handleIncomingRawMessage(data);
         };
         this.socket = socket;
+        window.addEventListener('offline', () => this.socket.reconnect());
```

Comment:
I like this change, but according to the documentation, this is not 100% reliable. Is it fine to be slightly more paranoid and also attach to `online` and only trigger `this.socket.reconnect()` if `this.socket.readyState` is neither `CONNECTING` nor `OPEN`? In an attempt to make it more reliable.

Reply:
I like that idea and implemented it like that. However, it doesn't make much sense to check for `OPEN` as the state can actually be `OPEN` while no network connection is established, which is actually the main reason behind this PR.